src/storage/recovery/recovery_manager.cpp

```diff
@@ -103,6 +224,14 @@ void RecoveryManager::ProcessCommittedTransaction(terrier::transaction::timestam
   txn_manager_->Commit(txn, transaction::TransactionUtil::EmptyCallback, nullptr);
 }
 
+bool RecoveryManager::IsCatalogTable(catalog::table_oid_t table_oid) {
+  return !table_oid == !catalog::postgres::DATABASE_TABLE_OID ||
+         !table_oid == !catalog::postgres::NAMESPACE_TABLE_OID || !table_oid == !catalog::postgres::CLASS_TABLE_OID ||
+         !table_oid == !catalog::postgres::COLUMN_TABLE_OID || !table_oid == !catalog::postgres::CONSTRAINT_TABLE_OID ||
+         !table_oid == !catalog::postgres::INDEX_TABLE_OID || !table_oid == !catalog::postgres::TYPE_TABLE_OID ||
+         !table_oid == !catalog::postgres::LANGUAGE_TABLE_OID || !table_oid == !catalog::postgres::PRO_TABLE_OID;
```

Comment:
There is a *much* easier check for this.  All catalog OIDs are below `START_OID` and all user-defined OIDs are greater than or equal to `START_OID`.